src/main/kotlin/snyk/amplitude/api/AmplitudeExperimentApiClient.kt

```diff
@@ -20,15 +20,15 @@ class AmplitudeExperimentApiClient private constructor(
      */
     @Suppress("TooGenericExceptionCaught")
     fun allVariants(user: ExperimentUser): Map<String, Variant> {
-        if (user.userId.isBlank()) {
-            log.warn("user id is empty; amplitude may not resolve identity")
+        if (user.userId.isBlank() && user.deviceId.isBlank()) {
```

Comment:
Why would we even try to fetch the variants if both properties are blank? This will lead to undefined behaviour and wouldn't count towards the experiment data.

Reply:
Because we still fetch it with a deviceID, so we don't need the userID.

Reply:
let me ask you differently - what's the purpose of fetching it with a blank device id and blank userId?

Reply:
The deviceID is never blank, as it is automatically populated on instance creation with the default value.